doctor.py

```diff
@@ -2,19 +2,96 @@
 
 import argparse
 from collections.abc import Sequence
+from concurrent.futures import ThreadPoolExecutor
```

Comment:
The documentation for both  [ThreadPoolExecutor](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor) and [ProcessPoolExecutor](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor) note that deadlocks can occur. In practice, since this is an API we'll be distributing to a wide audience with varying experience levels, it means deadlocks *will* occur.

Our fixer API needs to be more foolproof than that, IMO.

> Deadlocks can occur ...

Which in practice means deadlocks *will* occur.

Reply:
I don't think this is a concern for the TPE we're using - check authors are not gonna be able to reference other futures